ibis/backends/sql/compilers/snowflake.py

```diff
@@ -608,6 +608,12 @@ def visit_Quantile(self, op, *, arg, quantile, where):
         quantile = self.f.percentile_cont(quantile)
         return sge.WithinGroup(this=quantile, expression=order_by)
 
+    def visit_ApproxQuantile(self, op, *, arg, quantile, where):
+        if where is not None:
+            arg = self.if_(where, arg, NULL)
+
+        return self.f.approx_percentile(arg, quantile)
```

Comment:
I guess we can't chuck this into `SIMPLE_OPS`?

Reply:
No - similar to `group_concat` some backends get grumpy if `quantile` isn't a literal.